generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/MessageListenerCodeGenerator.java

```diff
@@ -111,21 +118,32 @@ TypeSpec generatedCodeFor(final Resource resource, final BaseUri baseUri, final
                             .build())
                     .addAnnotation(messageDrivenAnnotation(component, resource.getActions(), resourceUri, baseUri, listenToAllMessages));
 
-            addLoggerAndSchemaInterceptorsIfNotGeneralJsonMimeType(resource, typeSpecBuilder);
+            if (!containsGeneralJsonMimeType(resource.getActions())) {
+                typeSpecBuilder.addAnnotation(AnnotationSpec.builder(Interceptors.class)
+                        .addMember(DEFAULT_ANNOTATION_PARAMETER, "$T.class", JmsLoggerMetadataInterceptor.class)
+                        .addMember(DEFAULT_ANNOTATION_PARAMETER, "$T.class", JsonSchemaValidationInterceptor.class)
+                        .build());
+            }
+
+            if (shouldAddCustomPoolConfiguration(generatorConfiguration)) {
+                typeSpecBuilder.addAnnotation(AnnotationSpec.builder(Pool.class)
+                        .addMember(DEFAULT_ANNOTATION_PARAMETER, "$S", poolNameFrom(resourceUri, component))
+                        .build());
+            }
 
             return typeSpecBuilder;
         }
 
         throw new IllegalStateException(format("JMS Endpoint generation is unsupported for framework component type %s", component));
     }
 
-    private void addLoggerAndSchemaInterceptorsIfNotGeneralJsonMimeType(final Resource resource, final TypeSpec.Builder typeSpecBuilder) {
-        if (!containsGeneralJsonMimeType(resource.getActions())) {
-            typeSpecBuilder.addAnnotation(AnnotationSpec.builder(Interceptors.class)
-                    .addMember(DEFAULT_ANNOTATION_PARAMETER, "$T.class", JmsLoggerMetadataInterceptor.class)
-                    .addMember(DEFAULT_ANNOTATION_PARAMETER, "$T.class", JsonSchemaValidationInterceptor.class)
-                    .build());
-        }
+    private String poolNameFrom(final MessagingResourceUri resourceUri, final String component) {
+        return format("%s-%s-pool", resourceUri.hyphenated(), component.toLowerCase().replace("_", "-"));
+    }
+
+    private boolean shouldAddCustomPoolConfiguration(final GeneratorConfig configuration) {
+        return configuration.getGeneratorProperties().containsKey(CUSTOM_MDB_POOL_CONFIG_PROPERTY)
```

Comment:
Does the GET throw an exception or return null?

Generally speaking it is better to do the .equals() on the constant as it will never be a null pointer. Can this be simplified to "TRUE".equalsIgnoreCase(configuration.getGerenatorPropertions().get(CUSTOM_...) ?